server/handler/frontend.go

```diff
@@ -55,6 +61,164 @@ func LoadTemplates(c *FilesConfig, basePath string) (templatetree.HTMLTree, erro
 
 			return r
 		},
+		"getPredicateInfo": func(predicateInfo *common.PredicateInfo) []string {
```

Comment:
Because these new functions are quite complicated, I think they should be defined as top-level named functions rather than defined anonymously as part of this map.

Reply:
fixed by putting long functions at the end of the file.